packages/sanity/src/form/studio/inputResolver/inputResolver.tsx

```diff
@@ -106,10 +107,12 @@ function PrimitiveField(field: FieldProps) {
 }
 
 function ObjectOrArrayField(field: ObjectFieldProps | ArrayFieldProps) {
+  const level = getFieldLevel(field as ObjectFieldProps)
```

Comment:
Would be a tad safer to do a type assertion here:
```suggestion
  const level = isObjectField(field) ? getFieldLevel(field) : field.level
```
(You'll need to add this import as well: `import {isObjectField} from '../../utils/asserters'`)

Reply:
Thanks!

However, that won't pass array fields to the `getFieldLevel` helper. I think it needs to be something like this:

`const level = isObjectField(field) || isArrayField(field) ? getFieldLevel(field) : field.level`

But that results TS errors. Any advice?

Reply:
Maybe better to split them up, like `getObjectFieldLevel` and `getArrayFieldLevel`. And do something like:

```js
  const level = useMemo(() => {
    if (isObjectField(field)) {
      return getObjectFieldLevel(field)
    }

    if (isArrayField(field)) {
      return getArrayFieldLevel(field)
    }

    return field.level
  }, [field])

```

Reply:
>Maybe better to split them up, like getObjectFieldLevel and getArrayFieldLevel. And do something like:
> const level = isObjectField(field) ? getObjectFieldLevel(field) : getArrayFieldLevel(field)

Ah, yes, this would be the way to go - the TS errors you got was because array fields doesn't have a `fields` property, and that would actually become a runtime error with the earlier type cast.

Reply:
Updated! 👍